clients/js/src/auth.ts

```diff
@@ -130,6 +133,85 @@ class BasicAuthClientAuthProvider implements ClientAuthProvider {
     }
 }
 
+class TokenAuthCredentials implements AbstractCredentials<SecretStr> {
+    private readonly credentials: SecretStr;
+
+    constructor(_creds: string) {
+        this.credentials = new SecretStr(_creds)
+    }
+
+    getCredentials(): SecretStr {
+        return this.credentials;
+    }
+}
+
+export class TokenCredentialsProvider implements ClientAuthCredentialsProvider<TokenAuthCredentials> {
+    private readonly credentials: TokenAuthCredentials;
+
+    constructor(_creds: string | undefined) {
+        if (_creds === undefined && !process.env.CHROMA_CLIENT_AUTH_CREDENTIALS) throw new Error("Credentials must be supplied via environment variable (CHROMA_CLIENT_AUTH_CREDENTIALS) or passed in as configuration.");
+        this.credentials = new TokenAuthCredentials((_creds ?? process.env.CHROMA_CLIENT_AUTH_CREDENTIALS) as string);
+    }
+
+    getCredentials(): TokenAuthCredentials {
+        return this.credentials;
+    }
+}
+
+export class TokenClientAuthProvider implements ClientAuthProvider {
+    private readonly credentialsProvider: ClientAuthCredentialsProvider<any>;
+    private readonly providerOptions: { headerType: TokenHeaderType };
+
+    constructor(options: {
+        textCredentials: any;
+        credentialsProvider: ClientAuthCredentialsProvider<any> | undefined,
+        providerOptions?: { headerType: TokenHeaderType }
+    }) {
+        if (!options.credentialsProvider && !options.textCredentials) {
+            throw new Error("Either credentials provider or text credentials must be supplied.");
+        }
+        if (options.providerOptions === undefined || !options.providerOptions.hasOwnProperty("headerType")) {
+            this.providerOptions = {headerType: "AUTHORIZATION"};
+        } else {
+            this.providerOptions = {headerType: options.providerOptions.headerType};
+        }
+        this.credentialsProvider = options.credentialsProvider || new TokenCredentialsProvider(options.textCredentials);
+    }
+
+    authenticate(): ClientAuthResponse {
+        return new TokenClientAuthResponse(this.credentialsProvider.getCredentials(), this.providerOptions.headerType);
+    }
+
+}
+
+
+type TokenHeaderType = 'AUTHORIZATION' | 'X_CHROMA_TOKEN';
+
+const TokenHeader: Record<TokenHeaderType, (value: string) => { key: string; value: string; }> = {
+    AUTHORIZATION: (value: string) => ({key: "Authorization", value: `Bearer ${value}`}),
+    X_CHROMA_TOKEN: (value: string) => ({key: "X-Chroma-Token", value: value})
```

Comment:
maybe i missed it in the CIP, why do we have a separate `X-Chroma-Token` method?

Reply:
@jeffchuber, the reason is in case a secondary layer of auth happens at the proxy side, which may already be using the `Authorization` header.